lib/auth/register.go

```diff
@@ -152,6 +152,13 @@ func Register(params RegisterParams) (*proto.Certs, error) {
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
+		if !utils.IsEC2NodeID(params.ID.HostUUID) {
```

Comment:
Thanks for the IsEC2NodeID coverage, but I meant a test for the newly-added Register failure condition. WDYT?

Reply:
This is already a failure condition server-side, and there's a test for it https://github.com/gravitational/teleport/blob/b787b9579fe43ec6c82b2bcd8687d1b8eb7404eb/lib/auth/join_ec2_test.go#L314

I added this check client-side so that the error message isn't mixed in with other errors (trying auth & proxy). Just wanted to add a helpful message, not trying to spend a bunch of time on this, writing a unit test for ec2 joining at a high enough level to call `auth.Register` with a passing case would take me a minute, and I don't think the integration test (which only runs manually on ec2 atm) is the right place for this.